dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,587 @@
+        return  model
+
+    def correct_phase_wrap(self, phi, deg=False):
+
+        if deg:
+            return ((phi + 180.0) % 360.0) - 180.0
+        else:
+            return ((phi + np.pi) % (2.0 * np.pi)) - np.pi
+
+    def get_baselines(self, indexmap, inputmap):
+        
+        prod = defaultdict(list)
+        prodmap = defaultdict(list)
+        dist = defaultdict(list)
+        conj = defaultdict(list)
+        cyl = defaultdict(list)
+        scale = defaultdict(list)
+
+        # Compute feed positions without rotation
+        tools.change_chime_location(rotation=self.rot)
+        feedpos = tools.get_feed_positions(inputmap)
+        
+        for pp, (this_prod, this_conj) in enumerate(indexmap['stack']):
+
+            if this_conj:
+                bb, aa = indexmap['prod'][this_prod]
+            else:
+                aa, bb = indexmap['prod'][this_prod]
+            
+            inp_aa = inputmap[aa]
+            inp_bb = inputmap[bb]
+            
+            if not tools.is_chime(inp_aa) or not tools.is_chime(inp_bb):
+                continue
+                
+            if not self.include_auto and (aa == bb):
+                continue
+
+            if not self.include_intracyl and (inp_aa.cyl == inp_bb.cyl):
+                continue
+
+            this_dist = list(feedpos[aa, :] - feedpos[bb, :])
+
+            if tools.is_array_x(inp_aa) and tools.is_array_x(inp_bb):
+                key = 'XX'
+
+            elif tools.is_array_y(inp_aa) and tools.is_array_y(inp_bb):
+                key = 'YY'
+
+            elif not self.include_crosspol:
+                continue
+                
+            elif tools.is_array_x(inp_aa) and tools.is_array_y(inp_bb):
+                key = 'XY'
+
+            elif tools.is_array_y(inp_aa) and tools.is_array_x(inp_bb):
+                key = 'YX'
+
+            else:
+                raise RuntimeError("CHIME feeds not polarized.")
+
+            this_cyl = '%s%s' % (self.get_cyl(inp_aa.cyl), self.get_cyl(inp_bb.cyl))
+            if self.sep_cyl:
+                key = key + '-' + this_cyl
+
+            prod[key].append(pp)
+            prodmap[key].append((aa, bb))
+            conj[key].append(this_conj)
+            dist[key].append(this_dist)
+            cyl[key].append(this_cyl)
+
+            if aa == bb:
+                scale[key].append( 0.5 )
+            else:
+                scale[key].append( 1.0 )
+
+        for key in prod.keys():
+            prod[key] = np.array(prod[key])
+            prodmap[key] = np.array(prodmap[key], dtype=[('input_a', '<u2'), ('input_b', '<u2')])
+            dist[key] = np.array(dist[key])
+            conj[key] = np.nonzero(np.ravel(conj[key]))[0] 
+            cyl[key] = np.array(cyl[key])
+            scale[key] = np.array(scale[key])
+            
+        tools.change_chime_location(default=True)
+            
+        return prod, prodmap, dist, conj, cyl, scale
+
+    def get_cyl(self, cyl_num):
```

Comment:
Going to create a dias/utils/helpers for these helper functions that more than one analyzer re-use.